src/libraries/System.Diagnostics.Process/src/System/Diagnostics/AsyncStreamReader.cs

```diff
@@ -253,11 +253,20 @@ private bool FlushMessageQueue(bool rethrowInNewThread)
 
         // Wait until we hit EOF. This is called from Process.WaitForExit
         // We will lose some information if we don't do this.
-        internal void WaitUtilEOF()
+        internal void WaitUntilEOF()
         {
-            if (_readToBufferTask != null)
+            if (_readToBufferTask is Task task)
             {
-                _readToBufferTask.GetAwaiter().GetResult();
+                task.GetAwaiter().GetResult();
+                _readToBufferTask = null;
+            }
+        }
+
+        internal async ValueTask WaitUntilEOFAsync(CancellationToken cancellationToken)
+        {
+            if (_readToBufferTask is Task task)
+            {
+                await task.WithCancellation(cancellationToken).ConfigureAwait(false);
                 _readToBufferTask = null;
```

Comment:
We only want to clear this if the await on the task doesn't throw?

Reply:
It is not clear to me why the task should be cleared at all by the awaiting method.